app/controllers/blogs_controller.rb

```diff
@@ -1,73 +1,89 @@
 class BlogsController < ApplicationController 
-    def index
-        #@user = User.find_by(id:params[:user_id])
-        if(session[:user_id])
-            @user = User.find_by(id:session[:user_id])
-            @blog = @user.blogs.all
-        end
+  def all_blogs
+    @blog = Blog.all
+  end
+
+  def index
+    if !Current.user.nil?
+      @blog = Current.user.blogs.all
+    else 
+      redirect_to sign_in_path
     end
+  end
 
-    def show
-        @blog=Blog.find_by(id: params[:id])
+  def show
+    @blog=Blog.where(id: params[:id]).first
+    if @blog.nil?
+      flash[:alert]="No blogs"
+      redirect_to root_path
     end
+  end
 
-    def new
+  def new
+    if Current.user
       @blog = Blog.new
+    else 
+      redirect_to sign_in_path
     end
+  end
 
     def create
-        if(session[:user_id])
-            @user = User.find_by(id:session[:user_id])
-            @blog = @user.blogs.new(blog_params)
+      if Current.user
+        @blog = Current.user.blogs.new(blog_params)
 
-            if @blog.save
-                redirect_to @blog
-            end
-      #redirect_to user_blogs_path
+        if @blog.save
+          redirect_to @blog
         else
           render :new
         end
+      else
+        redirect_to sign_in_path
+      end
     end
 
     def edit
-        if(session[:user_id])
-            @user = User.find_by(id:session[:user_id])
-            @blog = @user.blogs.find_by(id:params[:id])
+      if Current.user
+        blog = Current.user.blogs.where(id: params[:id]).first
```

Comment:
how will this work?
`blog` variable is different than `@blog`
test code properly